### ISSUE TITLE
Evaluation on PascalVOC Dataset

### ISSUE THREAD

**Issue by Ekta246** [rwightman/efficientdet-pytorch#102]
Hi @rwightman super excited to work with the branch "more_datasets". It is helpful.

However, I was wondering, if you had validated the efficientdet-d0 model on the PascalVOC dataset. Are there any pre-trained weights based on the VOC dataset?
However, I am unable to use the coco_api on my custom dataset due to the many restrictions and the COCO format for the evaluation part of my custom dataset.
For the training, I had written a custom dataloader.

Looking forward to hearing from you.

**Comment by rwightman** [comment]
@Ekta246 I fine-tuned one VOC efficientdet_d0 model (from coco pretrained weights) while testing that branch, it was just a squeek under 80mAP. I didn't really search hard for hparams, it was quick and overfit early, etc so I wasn't planning to release it. At some point I'll likely add pretrained models for VOC and OpenImages via different model configs.

One of the goals of the dataset changes on that branch was to provide differing styles of datasets covering most important open datasets (COCO, VOC, OpenImages) and illustrating different styles (COCO JSON anno, XML anno, large multi-CSV). By spliting the parsers from the base, with similar interfaces and providing config dataclasses I hope that it's fairly easy to mix and match and create configs or parsers for other datasets to work with this...

At the very end I realized validation was an issue since I couldn't rely on coco_ap anymore. So I blitzed through some Google Tensorflow Models code and ported their Pascal and OpenImages evaluator. There are other simpler VOC evaluator routines, but it's hard to find OpenImages evalutors. You could likely use the Pascal evaluator as is for a custom dataset, or customize one with different args/subclass for the TfmEvaluator.

I've got another test training session going for the more_datasets branch. If it ends well I'll likely merge to master in the next few days and resolve remaining issues as they come up...

**Comment by rwightman** [comment]
For reference the cmd line I used for that fine tuning was:`/distributed_train.sh 4 /data/f/VOCdevkit --model efficientdet_d0 --dataset voc0712 -b 16 --amp --lr .005 --sync-bn --opt fusedmomentum --warmup-epochs 3 --lr-noise 0.4 0.9 --model-ema --model-ema-decay 0.9966 --epochs 200 --num-classes 20 --pretrained`

it was overfitting in the low 100s though, so hparams could use more tuning, this was on a system with 4 older Pascal Titan Xp/ Titan Pascal cards.

**Comment by Ekta246** [comment]
Hi,

Thanks alot @rwightman for such a clear and crisp explanation.
I see the Coco_api has some restrictions to the Coco datasets format. That
is why I evaluating on the Pascal VOC formats with ignoring the masks
labels while evaluation. However, I think I will be good

Thanks again!

**Comment by Ekta246** [comment]
Also, a very important question if the training needs to be done with the
bounding box coordinates(ymin,xmin,ymax,xmax) while parsing the data from
the dataloader? Or the code looks after the conversion from
(xmin,ymin,xmax,ymax) to (ymin,xmin,ymax,xmax)? I sense the latter is
correct.
Feel free to correct me or suggest me.

Thanks

**Comment by rwightman** [comment]
@Ekta246 the models expect yxyx as inputs for bbox targets, the example datasets are defaulted to output yxyx for that reason, but they do have a flag to flip it if other model types are used. 

Since I use PyTorch NMS, on the output I have to switch it to xyxy. On the master I was focused on COCO so I then converted to xywh by default. On the more_datasets branch I leave the output as xyxy and let the evaluator decide if it needs to be xywh, xyxy, or back to yxyx
